packages/sol-meta/src/contractMocker.ts

```diff
@@ -0,0 +1,123 @@
+import * as S from 'solidity-parser-antlr';
+
+import { makeConstructor, nonAbstractForcer } from './constructor';
+import { exposeNode } from './exposer';
+import { flattenContract } from './flattener';
+import { stubFunction } from './stubber';
+import { FunctionScript, scriptFunction } from './scripter';
+import { SourceCollection } from './source_reader';
+import * as utils from './utils';
+import { unparse } from './unparser';
+
+export const mockContractName = (contractName: string) => `${contractName}Mock`;
+
+export interface ContractMockOptions {
+    constructors: {
+        [parentName: string]: utils.Litteral[];
+    };
+    scripted: {
+        [functionName: string]: FunctionScript[];
+    };
+}
+
+export function mockContract(
+    sources: SourceCollection,
+    path: string,
+    contractName: string,
+    options: ContractMockOptions,
+): S.SourceUnit {
+    const sourceInfo = sources[path];
+
+    // Flatten target contract, this collapses the inheritane hierarchy
+    if (!(contractName in sourceInfo.contracts)) {
+        throw new Error(`Could not find contract ${contractName} in "${path}".`);
+    }
+    const [flat, parents] = flattenContract(sourceInfo.contracts[contractName], name => {
+        if (!(name in sourceInfo.scope)) {
```

Comment:
I will not comment on any built-in function vs lodash for now. Let's just talk with Fabio about it as he've mentioned above.